test/e2e-tests.sh

```diff
@@ -40,6 +40,21 @@ header "Setting up environment"
 install_pipeline_crd
 
 failed=0
+function alpha_gate() {
+  local gate="$1"
+  if [ "$gate" != "alpha" ] && [ "$gate" != "stable" ] && [ "$gate" != "beta" ] ; then
+    printf "Invalid gate %s\n" ${gate}
+    exit 255
+  fi
+  if [ "$gate" == "alpha" ] ; then
+    DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+    printf "Setting up environment for alpha features"
+    install_spire
+    patch_pipline_spire
+    kubectl apply -n tekton-pipelines -f "$DIR"/testdata/spire/config-spire.yaml
+    failed=0
+  fi
+}
```

Comment:
did you mean for this function to be about enabling spire? 🤔

Reply:
If we want to enable SPIRE we need to do a few changes to the pipelines deployment, in particular mount the spiffe-workload-api volume.
So for alpha tests this function makes all the changes to the deployment, which includes pipeline volume mounts and installing spire at the setup stage.
But it does not enable spire itself, i.e. it does not set the feature flag enforce_nonfalsifiability to spire, which ensures that all alpha tests except 2 in taskrun_test.go run without spire enabled.
For the 2 in taskrun_test.go we enable spire right in the test.

Reply:
yes, the name is misleading -- `add_spire` is better

Reply:
Done.